### PR TITLE
Adding changes to nargo compile

### LINKED ISSUE #26: Document `CompiledProgram` JSON
Issue:
- https://github.com/noir-lang/noir/issues/897

PR:
- https://github.com/noir-lang/noir/pull/922

`nargo compile` in v0.3.0 / newer Nightlies generates a `CompiledProgram` JSON file that contains the ACIR (as _circuit_) and the ABI (as _abi_).

### LINKED ISSUE #63: Document `CompiledProgram` JSON
Issue:
- https://github.com/noir-lang/noir/issues/897

PR:
- https://github.com/noir-lang/noir/pull/922

`nargo compile` in v0.3.0 / newer Nightlies generates a `CompiledProgram` JSON file that contains the ACIR (as _circuit_) and the ABI (as _abi_).

### PR DESCRIPTION
**Pull request by signorecello**
closes noir-lang/docs#26

### CODE REVIEW

**Review by Savio-Sou — APPROVED** [review]
Do we want to describe how is ABI formed / used?

LGTM either way. I'll let Tom confirm if there's anything else we should add.

File: src/getting_started/nargo/commands.md [hunk 1]
```diff
@@ -47,21 +47,12 @@ _Arguments_
 
 Generate a Solidity verifier smart contract for the program.
 
-## `nargo compile [FLAGS] <circuit_name>`
+## `nargo compile <circuit_name>`
 
-Compile the program and its secret execution trace into [ACIR](../../acir.md) format.
+Compile the program and its secret execution trace into a JSON file containing the ACIR representation, and the ABI of the circuit.
```
Comment (TomAFrench) [inline src/getting_started/nargo/commands.md:52]:
```suggestion
Compile the program and its secret execution trace into a JSON build artifact file containing the ACIR representation, and the ABI of the circuit. This build artifact can then be used to generate and verify proofs.
```

**Review by TomAFrench — APPROVED** [review]
I don't think we need to mention how the ABI is generated, I'd be explicit in calling this a build artifact however.

File: src/getting_started/nargo/commands.md [hunk 2]
```diff
@@ -47,21 +47,12 @@ _Arguments_
...
 
-- `<circuit_name>` - The name of the ACIR file
-- `[FLAGS]` - Add `--witness` to compile the witness file as well
+- `<circuit_name>` - The name of the circuit file
 
-_Usage_
-
-Running the command would create a new folder `build` with the compiled `<circuit_name>.acir` file in your project directory.
-
-To also compile a witness file, fill in the values in `Prover.toml` generated from `nargo check` and run the command with the `--witness` flag. A `<circuit_name>.tr` file would be compiled in the `build` folder.
```
Comment (jfecher) [inline src/getting_started/nargo/commands.md:63]: Why was the flags section removed? It seems the `--witness` flag is outdated, but we also have the `--circuits` flag now which could replace it

Comment (signorecello) [reply src/getting_started/nargo/commands.md:63]: Sorry Jake, this was merged already before your review… Shall we raise a new issue, or revert this PR?

Comment (jfecher) [reply src/getting_started/nargo/commands.md:63]: We can just make a new issue/pr

Comment (TomAFrench) [reply src/getting_started/nargo/commands.md:63]: The changes related to `witness` are made in https://github.com/noir-lang/book/pull/57

**Pull request merged**